This patch is based on a revert of upstream commit 5dc1d6ab3105739ae61039f422e5246f0eee4f64,
adapted for use in the Debian package by Peter Michael Green.

Index: indexmap/src/lib.rs
===================================================================
--- indexmap.orig/src/lib.rs
+++ indexmap/src/lib.rs
@@ -49,8 +49,8 @@
 //!
 //! [feature flags]: https://doc.rust-lang.org/cargo/reference/manifest.html#the-features-section
 //! [`no_std`]: #no-standard-library-targets
-//! [`Serialize`]: `::serde_core::Serialize`
-//! [`Deserialize`]: `::serde_core::Deserialize`
+//! [`Serialize`]: `::serde::Serialize`
+//! [`Deserialize`]: `::serde::Deserialize`
 //! [`BorshSerialize`]: `::borsh::BorshSerialize`
 //! [`BorshDeserialize`]: `::borsh::BorshDeserialize`
 //! [`borsh`]: `::borsh`
Index: indexmap/src/map/serde_seq.rs
===================================================================
--- indexmap.orig/src/map/serde_seq.rs
+++ indexmap/src/map/serde_seq.rs
@@ -9,7 +9,7 @@
 //!
 //! ```
 //! # use indexmap::IndexMap;
-//! # use serde::{Deserialize, Serialize};
+//! # use serde_derive::{Deserialize, Serialize};
 //! #[derive(Deserialize, Serialize)]
 //! struct Data {
 //!     #[serde(with = "indexmap::map::serde_seq")]
@@ -18,8 +18,8 @@
 //! }
 //! ```
 
-use serde_core::de::{Deserialize, Deserializer, SeqAccess, Visitor};
-use serde_core::ser::{Serialize, Serializer};
+use serde::de::{Deserialize, Deserializer, SeqAccess, Visitor};
+use serde::ser::{Serialize, Serializer};
 
 use core::fmt::{self, Formatter};
 use core::hash::{BuildHasher, Hash};
@@ -66,7 +66,7 @@ where
 ///
 /// ```
 /// # use indexmap::IndexMap;
-/// # use serde::Serialize;
+/// # use serde_derive::Serialize;
 /// #[derive(Serialize)]
 /// struct Data {
 ///     #[serde(serialize_with = "indexmap::map::serde_seq::serialize")]
@@ -119,7 +119,7 @@ where
 ///
 /// ```
 /// # use indexmap::IndexMap;
-/// # use serde::Deserialize;
+/// # use serde_derive::Deserialize;
 /// #[derive(Deserialize)]
 /// struct Data {
 ///     #[serde(deserialize_with = "indexmap::map::serde_seq::deserialize")]
Index: indexmap/src/serde.rs
===================================================================
--- indexmap.orig/src/serde.rs
+++ indexmap/src/serde.rs
@@ -1,10 +1,10 @@
 #![cfg_attr(docsrs, doc(cfg(feature = "serde")))]
 
-use serde_core::de::value::{MapDeserializer, SeqDeserializer};
-use serde_core::de::{
+use serde::de::value::{MapDeserializer, SeqDeserializer};
+use serde::de::{
     Deserialize, Deserializer, Error, IntoDeserializer, MapAccess, SeqAccess, Visitor,
 };
-use serde_core::ser::{Serialize, Serializer};
+use serde::ser::{Serialize, Serializer};
 
 use core::fmt::{self, Formatter};
 use core::hash::{BuildHasher, Hash};
Index: indexmap/Cargo.toml
===================================================================
--- indexmap.orig/Cargo.toml
+++ indexmap/Cargo.toml
@@ -56,7 +56,6 @@ rustdoc-args = [
 [features]
 default = ["std"]
 serde = [
-    "dep:serde_core",
     "dep:serde",
 ]
 std = []
@@ -118,8 +117,8 @@ default-features = false
 version = "1.9"
 optional = true
 
-[dependencies.serde_core]
-version = "1.0.220"
+[dependencies.serde]
+version = "1.0.219"
 optional = true
 default-features = false
 
@@ -147,11 +146,6 @@ version = "1.0"
 features = ["derive"]
 default-features = false
 
-[target."cfg(any())".dependencies.serde]
-version = "1.0.220"
-optional = true
-default-features = false
-
 [lints.clippy]
 style = "allow"
 
